lib/js_locale_helper.rb

```diff
@@ -134,14 +134,40 @@ def self.translations_for(locale_str)
     Marshal.load(Marshal.dump(translations))
   end
 
+  def self.output_MF(locale)
+    require "messageformat"
+
+    message_formats =
+      I18n.fallbacks[locale]
+        .each_with_object({}) do |l, hash|
+          translations = translations_for(l, no_fallback: true)
+          hash[l.to_s.dasherize] = remove_message_formats!(translations, l).merge(
+            TranslationOverride
+              .mf_locales(l)
+              .pluck(:translation_key, :value)
+              .to_h
+              .transform_keys { _1.sub(/^[a-z_]*js\./, "") },
+          )
+        end
+        .compact_blank
+    compiled = MessageFormat.compile(message_formats.keys, message_formats)
+
+    <<~JS
+      import Messages from '@messageformat/runtime/messages';
+      #{compiled.sub("export default", "const msgData =")};
+      const messages = new Messages(msgData, "#{locale.to_s.dasherize}");
+      messages.defaultLocale = "en";
+      globalThis.I18n._mf_messages = messages;
+    JS
+  end
```

Comment:
Writing this as a browser-native ESM module introduces all the extra complexity around importmap and `<script type="module"`

Could we make it a regular JS file which calls a function on the global I18n object? Something like:

```ruby
<<~JS
  globalThis.I18n.setMfData(#{compiled});
JS
```

and have all the `new Message()` stuff handled inside the [I18n module](https://github.com/discourse/discourse/blob/7c26d5d084ee6d5277efb1f713ec7306576c62fd/app/assets/javascripts/discourse-i18n/src/index.js#L11C16-L11C16) which is compiled by Webpack?